src/odemis/acq/fastem.py

```diff
@@ -195,10 +201,15 @@ def acquire(roa, path, scanner, multibeam, descanner, detector, stage, ccd, beam
 
     # TODO: pass path through attribute on ROA instead of argument?
     # Create a task that acquires the megafield image.
-    task = AcquisitionTask(scanner, multibeam, descanner, detector, stage, ccd, beamshift, lens, roa, path, f)
+    task = AcquisitionTask(scanner, multibeam, descanner, detector, stage, ccd, beamshift, lens, roa, path, f,
+                           pre_calibrate=pre_calibrate)
 
     f.task_canceller = task.cancel  # lets the future know how to cancel the task.
 
+    if pre_calibrate:
+        # NEW:
+        _executor.submitf(f, task.pre_calibrate)
```

Comment:
Does the executor take the order of calibration and acquisition into account? So that when you first submit the pre-calibrate and then the run, is that order kept?